csharp/ql/lib/semmle/code/csharp/security/dataflow/UnsafeDeserializationQuery.qll

```diff
@@ -367,13 +496,8 @@ private class DataContractJsonSerializerDeserializeMethodSink extends DataContra
   }
 }
 
-private class DataContractJsonSafeConstructorTrackingConfiguration extends SafeConstructorTrackingConfig
```

Comment:
Would this also apply to some of the previous PRs where private classes were removed rather than deprecated?

Reply:
No, not in general for private classes - in this case it is because the private class extends an abstract class (that is - it populates the abstract class with elements) which means that it is implicitly "public".
I have written "Do not extend this class." in SafeConstructorTrackingConfig. In some of the other "similar" PRs we actually went ahead and deleted the abstract class even though this might break compatibility.
I think we should do the same here as we are no longer referring to `SafeConstructorTrackingConfig`.

Reply:
@hvitved: Is that acceptable - we did something similar for other queries that used abstract classes in conjunction with data flow?
Alternatively we need to back the change backwards compatible with the abstract class (which is also possible), but I think we had a conversation about that we didn't want that.

Reply:
OK, let's leave it as-is.

Reply:
@hvitved : Please note that I deleted the deprecated abstract class. Did you intend to approve that change?